core/primitives/src/runtime/parameter_table.rs

```diff
@@ -1,23 +1,57 @@
 use super::config::{AccountCreationConfig, RuntimeConfig};
+use near_primitives_core::account::id::ParseAccountError;
 use near_primitives_core::config::{ExtCostsConfig, VMConfig};
 use near_primitives_core::parameter::{FeeParameter, Parameter};
 use near_primitives_core::runtime::fees::{RuntimeFeesConfig, StorageUsageConfig};
+use near_primitives_core::types::AccountId;
 use num_rational::Rational;
-use serde::de::DeserializeOwned;
-use std::any::Any;
 use std::collections::BTreeMap;
 
+/// Represents values supported by parameter config.
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq)]
+#[serde(untagged)]
+pub(crate) enum ParameterValue {
+    Null,
```

Comment:
Most likely not for this PR, but I do wonder in general: Does it make sense to have a `Null` value in parameter definitions? I think today we need it to express "this parameter has been removed". But I am not convinced that we actually need a way to remove parameters, nor am I convinced that a Null value is the best way to do it.

Reply:
That's a good observation! We don't actually need an explicit `Null` value here during parsing, and can instead use `Option<ParameterValue>` in `ParameterTableDiff`. I will do this refactoring now.

For the second part of the question, I don't know whether we need to remove parameters, but we definitely need to be able to define them, so we need an `Optional` at least on one of the sides.
This touches a bit the work in https://github.com/near/nearcore/issues/7642, where I needed to revise some of these decisions, so we probably can discuss it separately on that issue.

Reply:
I got rid of `ParameterValue::Null` and rewrote `apply_diff` in non-trivial way as a part of this, PTAL.